exercises/codec-server/solution/codec-server/main.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+
+	edu_converters_go_code "github.com/temporalio/edu-converters-go-code/exercises/codec-server/solution"
+
+	"go.temporal.io/sdk/converter"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/tag"
+)
+
+var logger log.Logger
+
+// newCORSHTTPHandler wraps a HTTP handler with CORS support
+func newCORSHTTPHandler(web string, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", web)
+		w.Header().Set("Access-Control-Allow-Credentials", "true")
+		w.Header().Set("Access-Control-Allow-Headers", "Authorization,Content-Type,X-Namespace")
+
+		if r.Method == "OPTIONS" {
+			return
+		}
+
+		next.ServeHTTP(w, r)
+	})
+}
+
+// HTTP handler for codecs.
+// This remote codec server example supports URLs like: /{namespace}/encode and /{namespace}/decode
+// For example, for the default namespace you would hit /default/encode and /default/decode
+// It will also accept URLs: /encode and /decode with the X-Namespace set to indicate the namespace.
+func newPayloadCodecNamespacesHTTPHandler(encoders map[string][]converter.PayloadCodec) http.Handler {
```

Comment:
This is a pretty advanced codec server that has namespace-specific codecs and such. I question whether such advanced needs are necessary for learning compared to something simple more like https://github.com/temporalio/samples-go/blob/main/encryption/codec-server/main.go (but maybe still with cors wrapper).

Reply:
It is fairly advanced -- but if you check out https://github.com/temporalio/edu-converters-go-content/blob/using-a-codec-server/using-a-codec-server/sharing-converter-logic.md, I start with a smaller example, and only add CORS stuff, etc., as the example develops. I wanted to present enough an example that could start small but wouldn't have to be refactored to add integration features.

Reply:
CORS stuff makes sense, it is the namespace-specific codecs that I think may be unnecessarily advanced. Just one codec is probably good to demonstrate.